### PR TITLE
Backport 0.18 PRs #14451, #14686, #14564, #14568,#15063, #15411: deprecate BIP70 and build GUI without BIP70 support

### PR DESCRIPTION
**Pull request by dzutte-cpp**

### CODE REVIEW

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
pls see https://github.com/UdjinM6/dash/commit/886d25950a7b1bc1232c8d3e443f5a9a518c77fb and https://github.com/UdjinM6/dash/commit/83ce83e0d2a9df2b9ceba2cdf18038f1c23bd395

File: ci/matrix.sh [hunk 1]
```diff
@@ -55,7 +55,7 @@ elif [ "$BUILD_TARGET" = "win64" ]; then
   export DIRECT_WINE_EXEC_TESTS=true
 elif [ "$BUILD_TARGET" = "linux32" ]; then
   export HOST=i686-pc-linux-gnu
-  export BITCOIN_CONFIG="--enable-zmq --enable-glibc-back-compat --enable-reduce-exports --enable-crash-hooks LDFLAGS=-static-libstdc++"
+  export BITCOIN_CONFIG="--enable-zmq --disable-bip70 --enable-glibc-back-compat --enable-reduce-exports --enable-crash-hooks LDFLAGS=-static-libstdc++"
```
Comment (dzutte-cpp) [inline ci/matrix.sh:58]: Should we update other targets in this file, e.g., `linux64` right below, to include the `--disable-bip70` option too?

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

File: ci/matrix.sh (hunk 1 above)
Comment (UdjinM6) [reply ci/matrix.sh:58]: Having `--disable-bip70` in `linux32` job only follows 15411's logic, should be enough imo.

File: src/Makefile.am [hunk 2]
```diff
@@ -800,13 +800,14 @@ if HARDEN
 	$(AM_V_at) READELF=$(READELF) OBJDUMP=$(OBJDUMP) $(PYTHON) $(top_srcdir)/contrib/devtools/security-check.py < $(bin_PROGRAMS)
 endif
 
-
```
Comment (PastaPastaPasta) [inline src/Makefile.am:803]: 14451: this should be reverted

File: src/qt/paymentserver.cpp [hunk 3]
```diff
@@ -408,6 +297,10 @@ void PaymentServer::handleURIOrFile(const QString& s)
         QUrlQuery uri((QUrl(s)));
         if (uri.hasQueryItem("r")) // payment request URI
         {
+#ifdef ENABLE_BIP70
+            Q_EMIT message(tr("URI handling"),
+                tr("You are using a BIP70 URL which will be unsupported in the future."),
+                CClientUIInterface::ICON_WARNING);
```
Comment (PastaPastaPasta) [inline src/qt/paymentserver.cpp:303]: 14451: This should be reverted / commented out. We have no intention of removing BIP70 support.

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

File: src/qt/paymentserver.cpp (hunk 3 above)
Comment (dzutte-cpp) [reply src/qt/paymentserver.cpp:303]: Let's comment it out, should hopefully be better for future backports and cherry-picks to produce less conflicts.

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merge commit

**Pull request merged**